### DISCUSSION TITLE
[indexer] Get ready for nearcore 1.21.0

### DISCUSSION THREAD

**Discussion by khorolets** [near/nearcore#4689] [Node Public Interfaces]
`nearcore` is planned to be updated to version 1.21.0:
 - *testnet* on ~~Mon 16 Aug, 2021~~ UPD: Aug 18, 2021
 - *mainnet* on ~~Mon 23 Aug, 2021~~ UPD: Sep 6, 2021

Protocol upgrade is expected in this release (it means that your old indexers will work for ~2 days after release on testnet and ~7 days after release on mainnet)

### Quick info

**Commit hash** [4c4f18b581b7a652388c8a9c650e203726aabf7d](https://github.com/near/nearcore/releases/tag/1.21.0-rc.1)
**Expected issues** none
**Reference** have a look at how [we made it on NEAR Indexer for Explorer](https://github.com/near/near-indexer-for-explorer/pull/150) + https://github.com/near/near-indexer-for-explorer/pull/151

#### Introduction of strictly typed`AccountId` (#4440)

For your information nearcore 1.21.0 includes breaking change in primitive type `AccountId`. It used to be an alias to `String` but now it is a separate type. 

If your data handlers contain any logic around accounts (`receiver_id`, `signer_id`, etc.) you'll need to update the logic. 

For example, if you were watching for `Receipts` and check if the `receiver_id` corresponds to the name of the contract you're watching you might have had something like:

```rust
if &receipt.receiver_id == "mycoolcontract.near" {
...
}
```

Now `&receipt.receiver_id` has the type `&near_indexer::near_primitives::types::AccountId`. You may want to change your code as following:

```rust
if receipt.receiver_id.as_ref() == "mycoolcontract.near" {
...
}

OR

if receipt.receiver_id == near_indexer::near_primitives::types::AccountId::from_str("mycoolcontract.near").expect("account id is invalid") {
...
}
```

Please note the examples are synthetic.

#### Names and number of arguments of `init_config` has changes in `nearcore` (#4427 and #4453)

- `download` argument has been replaced with `download_config` and `download_genesis`
- `boot_nodes` argument was added
- `download_config_url` was added

Feel free to leave your questions in the comments below.

**Comment by bowenwang1996** [comment]
FYI the release is delayed

**Comment by khorolets** [comment]
FYI I've added a new commit hash and additional PR for reference.

**Comment by khorolets** [comment]
Heads up, release dates were updated
